submariner/OVN-Interconnect.md

```diff
@@ -0,0 +1,129 @@
+
+In this proposal, we explore two options to make Submariner work with OVN IC. More details about the upgrade and other design details shall
+be added once agree on one of them.
+
+### Option1: Add extra routes in existing Architecture
+
+ This proposal makes minimal changes in the network plugin syncer to get OVN IC working. With this, we add new routes in the OVN cluster
+router.
+
+![UpdatedTopology](./images/submariner-ovn-ic.png)
+
+These are the changes that will be needed
+
+1) Network plugin Syncer needs to be Zone aware
+2) In the Zone where Submariner Gateway is present an extra route will be added to direct the traffic to remote CIDRS connected via
+submariner to the Submariner router. This is for the remote traffic coming from other zones.
+3) In all the other zones a route will be added to direct the traffic to remote cluster CIDRS, connected via submariner, to the transit
+switch that connects to the gateway node
+
+#### Node Listener
+
+A new node listener shall be added in NetworkPluginSyncer.
+
+* The node listener shall be responsible for reading the node labels/annotations and identifying the zone in which the Submariner gateway is
+deployed and the transit switch IP for it.
+* When a new node is added or a node joins a zone, the node listener needs to call the NetworkPluginSyncer handler to program the required
+rules. It will also remove any existing stale rules.
+* When a node leaves no action is needed
+* When the gateway nodes join a new zone, leaving the current one, all the switches will need to be reprogrammed as the transit switch IP
+can change.
+
+#### Network Plugin Syncer
+
+Based on the info retrieved by the Node listener, the network plugin syncer can provision the Submariner infrastructure and program the
+rules in the OVN database.
+
+* If no zone is detected(will be required for backward compatibility) or a global zone is detected the Submariner infrastructure and flows
+will remain as it is today.
+
+* If a zone configuration is detected
+
+1) In the gateway node zone a new routing rule can be added to forward the traffic destined for Submariner remote CIDRS
+to the Submariner router. This is required for the traffic coming from pods in a node that is in a different zone.
+
+2) In the non-gateway zones a route will be added for the traffic destined for remote cluster CIDR to forward it to the gateway node
+transit switch.
+
+#### Pros of Option1
+
+This makes the least changes than option 2 and can be delivered within the time frame we have.
+
+#### Cons of Option1
+
+It depends on OVN labels/annotations for programming the routes, which means a change in these would break Submariner. Also given the plans
+to change the OVN -Submariner architecture this has to be reworked in future.
+
+### Option2: Submariner OVN Re-architecture
+
+There is an effort to optimize Submariner OVN integration. As a part of this Submariner Router and related ports and join switch shall be
```

Comment:
I'm not sure i understand this option. Are you saying that we should have a CR and the associated controller as part of OVN CNI? Will the OVNK team be okay to maintain the controller?

Reply:
This idea was proposed by the OVN team. The CR should be a generic one and not tied to Submariner.  Like providing a way to add a static route or router policy with the next hop and priority in a router. Also adding new ports as needed.

We will have to design this CRD and get it approved and implement the controller there.